### PR TITLE
hexfiend: init at 2.16.0

### PR DESCRIPTION
**Pull request by eliandoran**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Janik-Haag — COMMENTED** [review]
Code looks good, but I don't have a darwin machine so I can't test if it builds and works.

File: pkgs/os-specific/darwin/hexfiend/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchurl, undmg, ... }:
+
+stdenv.mkDerivation rec {
+  version = "2.16.0";
+  pname = "hexfiend";
+
+  src = fetchurl {
+    inherit version;
+    url = "https://github.com/HexFiend/HexFiend/releases/download/v${version}/Hex_Fiend_2.16.dmg";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/hexfiend/default.nix:9]:
```suggestion
    url = "https://github.com/HexFiend/HexFiend/releases/download/v${version}/Hex_Fiend_${lib.versions.majorMinor version}.dmg";
```

File: pkgs/os-specific/darwin/hexfiend/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchurl, undmg, ... }:
+
+stdenv.mkDerivation rec {
+  version = "2.16.0";
+  pname = "hexfiend";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/hexfiend/default.nix:5]:
```suggestion
  pname = "hexfiend";
  version = "2.16.0";
```

File: pkgs/os-specific/darwin/hexfiend/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchurl {
+    inherit version;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/hexfiend/default.nix:8]:
```suggestion
```

File: pkgs/os-specific/darwin/hexfiend/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchurl, undmg, ... }:
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/hexfiend/default.nix:1]:
```suggestion
{ stdenv, lib, fetchurl, undmg }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by Janik-Haag** [comment]
Some one in the macos matrix channel said it works fine so as soon as you have added the imrpovments sandro suggested this is good to be merged (atleast in my personal opinion)

**Comment by eliandoran** [comment]
@SuperSandro2000 , thank you for the thorough review. Everything should be fine now.

@SuperSandro2000 , @Janik-Haag , please re-review.

**Review by Janik-Haag — APPROVED** [review]
LGTM; you could add a meta.changelog attribute linking to https://hexfiend.github.io/HexFiend/ReleaseNotes.html

**Comment by eliandoran** [comment]
> LGTM; you could add a meta.changelog attribute linking to https://hexfiend.github.io/HexFiend/ReleaseNotes.html

Good point. Added!

**Pull request merged**